openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestCaseRepository.java

```diff
@@ -256,23 +267,84 @@ private void setTestSuiteSummary(TestCase testCase, Long timestamp, TestCaseStat
     for (TestSuite testSuite : testSuites) {
       testSuite.setSummary(null); // we don't want to store the summary in the database
       List<ResultSummary> resultSummaries = listOrEmpty(testSuite.getTestCaseResultSummary());
-      if (resultSummaries.isEmpty()) {
-        resultSummaries.add(resultSummary);
-      } else {
-        // We'll remove the existing summary for this test case and add the new one
+      if ((isDeleted) && (resultSummaries.isEmpty())) {
+        continue; // if we try to delete the state but not state is set then nothing to do
+      }
+
+      ResultSummary storedResultSummary = findMatchingResultSummary(resultSummaries, resultSummary.getTestCaseName());
+
+      if (!shouldUpdateResultSummary(storedResultSummary, timestamp, isDeleted)) {
+        continue; // if the state should not be updated then nothing to do
+      }
+
+      if (storedResultSummary != null) {
         resultSummaries.removeIf(summary -> summary.getTestCaseName().equals(resultSummary.getTestCaseName()));
+      }
+
+      if (!isDeleted) {
```

Comment:
do we need this condition , given we already checked it above?

Reply:
correct, it should be removed in the method